begin dashboard beta docs (#1660)
semgrep/semgrep-docs: docs/semgrep-appsec-platform/dashboard-beta.md

## Production backlog

This pane displays analytics related to findings detected in your primary or default branch. This typically means that the finding, usually a security issue, has made it to production environments.

### Key metrics 

| Key metrics    | Description |
| -------------- | ------ |
| Total new      | |
| Total fixed    | |
| Total ignored  | |
| Total net new  | |

### Charts

| Chart | Description |
| -------  | ------ |
| Open backlog         | Open findings over the chosen time period. This tracks the total findings from each scan and displays them. Lower values are better. |
| Backlog activity | Displays the number of new, net new, fixed, and ignored findings. A greater **Fixed** value is better. |

## Secure guardrails

This provides an overview of how secure guardrails in **PR or MR comments** are used in your organization. Other guardrail interfaces, such as the IDE or `pre-commit`, are not counted in this section.

![Secure guardrails pane](/img/dashboard-guardrails.png)
_**Figure**. Secure guardrails pane._

### Key metrics

| Key metrics    | Description |
| -------------- | ------ |

> | Findings shown to devs      | Percentage of findings shown to devs in PR or MR comments against the total findings count. An upward or stable trend is better. |

Author: > ...against the total findings count...

Wanted to confirm that the count of  'total findings' is the whole percentage?

:::

![Dashboard (beta) page](/img/dashboard-fold.png)
_**Figure**. The dashboard (beta_) page._

The dashboard is divided into several panes:

<dl>
<dt>Filters and configuration</dt>
<dd>Sets the time period, product, and project scope of the <strong>production backlog</strong> and <strong>secure guardrails</strong> charts in the page.</dd>
<dt>Production backlog</dt>
<dd>Displays data about all the findings detected in your primary or default branch.</dd>
<dt>Secure guardrails</dt>
<dd>Displays data relevant to the deployment of secure guardrails.</dd>
<dt>Most findings by project</dt>
<dd>Lists findings by project, grouped by **product** or **severity**.</dd>
<dt>Mean time to remediate</dt><!-- Median open finding age -->
<dd>A graph showing the time to remediate over the given time period.</dd>
</dl>

## Filters and configuration

Use the filters to gain a top-level view or zoom in to a specific period of time. Create quarterly overviews or recent incident statements for various AppSec stakeholders by specifying the following filters:

- Time period
- Project (repository)
- Semgrep product or type of scan (SAST, SCA, or Secrets)

> Configurations set here apply to the following panes:

Author: Hi @milanwilliams, does the filter pane apply to all charts, **including:** 

- Most findings by project
- Mean time to remediate? 

Wanted to be sure before adding them in. Thank you.

Reviewer: @s-santillan yes they do!!

- Production backlog
- Secure guardrails

### Recommended priority filters

By default, **<i class="fa-solid fa-toggle-large-on"></i> Recommended priority** filters are enabled. These filters display any finding that is **Critical** or **High** severity in **addition** to being:

- High confidence - if the finding is from Semgrep Code.
- Reachable - if the finding is from Semgrep Supply Chain.
- Valid - if the finding is from Secrets.

tk add links to definitions

## Production backlog

This pane displays analytics related to findings detected in your primary or default branch. This typically means that the finding, usually a security issue, has made it to production environments.

### Key metrics 

| Key metrics    | Description |
| -------------- | ------ |
| Total new      | |
| Total fixed    | |
| Total ignored  | |

> | Total net new  | |

Author: How is **Total net new** computed as opposed to **Total new**?

Reviewer: total net new = total new - (total amount of fixed + total amount of ignored). we can probably add some context about the goal being to get total net new as close to zero as possible, meaning that you're addressing things as quickly as new things are coming in

Author: Thank you! 
![image](https://github.com/user-attachments/assets/00a58cc3-58e3-4d45-9432-7b0d5cdbdb84)

## Secure guardrails

This provides an overview of how secure guardrails in **PR or MR comments** are used in your organization. Other guardrail interfaces, such as the IDE or `pre-commit`, are not counted in this section.

![Secure guardrails pane](/img/dashboard-guardrails.png)
_**Figure**. Secure guardrails pane._

### Key metrics

| Key metrics    | Description |
| -------------- | ------ |
| Findings shown to devs      | Percentage of findings shown to devs in PR or MR comments against the total findings count. An upward or stable trend is better. |
| Findings fixed before backlog      | Displays the percentage of findings that were fixed before they could be detected in a default branch or production backlog against the total findings count. An upward or stable trend is better. |

### Charts

| Chart | Description |
| -------  | ------ |
| Secure guardrails adoption  | Percent of new findings shown to developers over the specified time period. An upward or stable trend is better. |
| Guardrails activity | This chart displays a breakdown of the status of findings shown to developers; whether they were ignored, fixed, or remained open. A greater **Fixed** value is better. |

> ## Most findings by project

Author: @milanwilliams Why do we call it "Most findings by project" rather than "Findings by project"?

Reviewer: we want to surface the projects with the _most_ findings so users can figure out where to pay attention / allocate resources, so we labeled it "most" to emphasize the ordering is most -> least (rather than just an alphabetical list of projects or something)

Reviewer: i wonder if we should add something in about this surfacing "hotspots" - right now, we say that lower values are better, but i think we want to focus on the fact that they should pay attention and prioritize the projects with the most findings cc @milanwilliams

Author: Added the ff callout:
![image](https://github.com/user-attachments/assets/20e5bd91-328e-4788-bda7-56475e036b1f)

---
slug: dashboard-beta
title: Dashboard (beta)
hide_title: true
description: Use the Semgrep dashboard (beta) to gain an overview of your organization's security posture, including the deployment of guardrails.
tags:
  - Semgrep AppSec Platform
  - Secure guardrails
---

# Dashboard (beta)

> The Semgrep dashboard (beta) is an overview of your organization’s security posture from data aggregated within Semgrep AppSec Platform. It helps you achieve the following goals:

Reviewer: do we think it's necessary to call out the differences between this and the current dashboard? or what this offers vs the current dashboard? this may not be the right place, but i think it's an importance resource for us to have cc @milanwilliams

Author: I wrote out which parts are exclusive to the beta in the beta callout. I didn't make it too long because that's more or less the content of the doc and would be redundant. 
![image](https://github.com/user-attachments/assets/38e48dad-7a10-4570-8422-46e1fe6c0035)

- Evaluation of the success of your AppSec program, enabling you to know your current security risk.
  - This includes assessing the deployment of secure guardrails to your organization.
- Awareness of trends and opportunities that you can use to improve your security posture.
  - You are able to view **recurring security issues**, consequently taking action on them.

<!--
The dashboard provides the following features:

- Filters that enable you to track key metrics over a period of time, per Semgrep product, and per project.
-->

:::note
This feature is in a **private beta**. To request access, contact your Technical Account Manager or your Account Executive and let them know you'd like to join the dashboard beta.
:::

![Dashboard (beta) page](/img/dashboard-fold.png)
_**Figure**. The dashboard (beta_) page.

> The dashboard is divided into several panes:

Reviewer: i think this is prob a good opportunity for us to add some color on why these charts are important / what they help folks answer? cc @milanwilliams

Reviewer: some thoughts (mostly pulled from a slack post from milan, but would use her opinion as source of truth!):

**Production backlog:** 
This section can help answer the following questions: 
- How is my security posture doing over time? 
- Is my backlog decreasing or increasing? 
- Is my net change increasing/decreasing? 
- Are we addressing findings faster than new findings are coming in? 

**Secure guardrails:**
This section can help answer the following questions: 
- How many vulnerabilities did Semgrep prevent from hitting production, over time?
- Am I effectively introducing guardrails to my developers?  
- Of the issues shown to developers, are things being fixed or are they being ignored? 

**Most findings by project:**
This section can help answer the following questions: 
- Which of my projects have the most findings in a particular product area?
- Which of my projects have the most findings for a particular severity?

 **Median open finding age:**
This section can help answer the following question: 
- What is the amount of time a finding remains open, by product or by severity?

cc @milanwilliams

Author: This is super helpful and I believe the money shot for the doc. Thank you @sims-e

<dl>

> <dt>Filters and configuration</dt>

Reviewer: i wonder if we want to call out that you can view metrics for a single product, or choose to view metrics for all products? the experience differs slightly, in that when you're viewing a single product, you won't be able to group by product. this may be a nuance we don't want to call out, but just wanted to flag it!

Author: Added!
![image](https://github.com/user-attachments/assets/d3d48481-a93e-4e76-9e38-be042e310d5e)

<dd>Sets the time period, product, and project scope of the <strong>production backlog</strong> and <strong>secure guardrails</strong> charts in the page.</dd>
<dt>Production backlog</dt>

> <dd>Displays data about all the findings detected in your primary or default branch.</dd>

Reviewer: @s-santillan for both the production backlog and secure guardrails, i wonder if we should add info on the high-level metrics? the new/fixed/ignored is kind of self explanatory, but total net new in production backlog and the two metrics under guardrails could be helpful to call out?

Reviewer: ah ignore me, i see you do that later on :)

> <dt>Secure guardrails</dt>

Reviewer: @s-santillan should we define secure guardrails/link to secure guardrails docs?

Author: Ya, currently I can't add the link as the doc for secure guardrails isn't live, but can do it once the secure guardrails pr is merged.

Author: Secure guardrails link added - resolving.

<dd>Displays data relevant to the deployment of secure guardrails.</dd>
<dt>Most findings by project</dt>
<dd>Lists findings by project, grouped by **product** or **severity**.</dd>

> <dt>Mean time to remediate</dt><!-- Median open finding age -->

Reviewer: @s-santillan i think we shifted this to be median open finding age - so we can update this to something like: 

A graph showing the middle age of all Open findings, grouped by product or severity. Half of the open findings are older than this age, and half are newer.

<dd>A graph showing the time to remediate over the given time period.</dd>
</dl>

## Filters and configuration

Use the filters to gain a top-level view or zoom in to a specific period of time. Create quarterly overviews or recent incident statements for various AppSec stakeholders by specifying the following filters:

- Time period
- Project (repository)
- Semgrep product or type of scan (SAST, SCA, or Secrets)

Configurations set here apply to the following panes:

- Production backlog
- Secure guardrails

### Recommended priority filters

> By default, **<i class="fa-solid fa-toggle-large-on"></i> Recommended priority** filters are enabled. These filters display any finding that is **Critical** or **High** severity in **addition** to being:

Reviewer: i would maybe add a blurb that says if recommended priority filter is off, we show _all_ findings. users can choose to filter severity / confidence / reachability / validation themselves in the filters drawer cc @milanwilliams

<!-- vale on -->

### Charts

| Chart | Description |
| -------  | ------ |
| Secure guardrails adoption  | Percent of new findings shown to developers over the specified time period. An upward or stable trend is better. |
| Guardrails activity | This chart displays a breakdown of the status of findings shown to developers; whether they were ignored, fixed, or remained open. A greater **Fixed** value is better. |

## Most findings by project

A table providing a breakdown of findings for each project, grouped by product or severity. Lower values are better.

![Findings by project pane.](/img/dashboard-findings-by-project.png#sm-width-noborder)
_**Figure**. Findings by project; group them by product or severity._

## Mean time to remediate

<!-- check if title changes -->

![Mean time to remediate a finding.]/img/dashboard-mean-time-remediate.png)
_**Figure**. Mean time to remediate a finding._

A chart displaying the mean time to remediate a finding over the specified time period. Lower is better.

For a finding to be remediated, it must have any of the following statuses:

- Fixed

> - Ignored

Author: Not sure sure about Ignored, Fixing, and Reviewing triage statuses counting as "remediation."

Author: Fixing / Reviewing - considered open.

---
slug: dashboard-beta
title: Dashboard (beta)
hide_title: true
description: Use the Semgrep dashboard (beta) to gain an overview of your organization's security posture, including the deployment of guardrails.
tags:
  - Semgrep AppSec Platform
  - Secure guardrails
---

# Dashboard (beta)

> The Semgrep dashboard (beta) is an overview of your organization’s security posture from data aggregated within Semgrep AppSec Platform. It helps you:

Reviewer 2: ```suggestion
The Semgrep dashboard (beta) is an overview of your organization’s security posture based on data aggregated within Semgrep AppSec Platform. It helps you:
```

- Evaluate your AppSec program, enabling you to know your current security risk.

>   - The dashboard beta includes assessing the deployment and adoption of **[secure guardrails](/secure-guardrails/secure-guardrails-in-semgrep)** to your organization.

Reviewer 2: ```suggestion
  - The dashboard (beta) assesses the deployment and adoption of **[secure guardrails](/secure-guardrails/secure-guardrails-in-semgrep)** to your organization.
```

Author: maybe "provides metrics to help you assess" or something like that because by itself the dashboard doesn't assess anything, the person does.

- Become aware of trends and opportunities that you can use to improve your security posture.

> - Quickly filter data granularly for all the charts in the page as well as view priority findings.

Reviewer 2: ```suggestion
- Quickly filter data granularly for all the charts on the page and view priority findings.
```

:::note

> This feature is in a **private beta**. To request access, contact your Technical Account Manager or your Account Executive and let them know you'd like to join the dashboard beta.

Reviewer 2: ```suggestion
This feature is in **private beta**. To request access, contact your Technical Account Manager or your Account Executive and let them know you'd like to join the dashboard beta.
```

Beta-exclusive features include:

- Improved filtering, including a [recommended priority](#recommended-priority) set of filters to bring to your attention the most important findings quickly.
- Graphs and charts that track:
    - Developer engagement through secure guardrails.

>     - Changes in the number of open findings in production backlog, which lets you know if your team is addressing findings or if they are decreasing over time.

Reviewer 2: ```suggestion
    - Changes in the number of open findings in the production backlog, which lets you know if your team is addressing findings or if the number of new findings is decreasing over time.
```

    - The average time it takes to close a finding.
:::

![Dashboard (beta) page](/img/dashboard-fold.png)
_**Figure**. The dashboard (beta) page. Hover over the charts to view data for that point in time._

## Dashboard overview

The dashboard is divided into several panes:

<table>
<thead>
<tr>

> <td>Pane</td>

Reviewer 2: ```suggestion
<td><b>Pane</b></td>
```

> <td>Description</td>

Reviewer 2: ```suggestion
<td><b>Description</b></td>
```

</tr>
</thead>
<tbody>
<tr>
  <td>Today's recommended priority findings</td>
  <td>Provides a count of [priority](#recommended-priority) findings for all of your Semgrep products. Clicking any of the values takes you to the product's Findings page with those filters applied.</td>
</tr>
<tr>
<td>Filters and configuration</td>
<td>
Sets the filters for all the data in the page except for the **Today's recommended priority findings** pane.
</td>
</tr>
<tr>
<td>Production backlog</td>
<td>
Displays data about all the findings detected in your primary or default branch and helps you answer the following questions:
- How is my security posture doing over time?
- Is my backlog decreasing or increasing?
- Is the team addressing findings faster than new findings are coming in?
</td>
</tr>
<tr>
<td>[Secure guardrails](/secure-guardrails/secure-guardrails-in-semgrep)</td>
<td>
Displays data relevant to the deployment and adoption of secure guardrails. It helps address the following:

> - How many vulnerabilities did Semgrep prevent from entering production, over time?

Reviewer 2: ```suggestion
- How many vulnerabilities did Semgrep prevent from entering production over time?
```

### Key metrics

<!-- vale off -->
| Key metrics    | Description |
| -------------- | ------ |
| Findings shown to devs      | Number of findings shown to developers in PR or MR comments (the numerator) against the total findings count (denominator). An upward or stable trend is better. |
| Findings fixed before backlog      | Number of findings that were fixed before they could be detected in a default branch or production backlog (numerator) against the total findings count in the specified time period. An upward or stable trend is better. |

<!-- vale on -->

### Charts

| Chart | Description |
| -------  | ------ |
| Secure guardrails adoption  | Percent of new findings shown to developers over the specified time period. An upward or stable trend is better. |
| Guardrails activity | This chart displays a breakdown of the status of findings shown to developers; whether they were ignored, fixed, or remained open. A greater **Fixed** value is better. |

## Most findings by project

A table listing projects from most open findings to least, grouped by product or severity. Lower values are better.

:::tip

> It is recommended to prioritize triage and remediation on the top projects listed in this table, especially if the priority filters are enabled.

Reviewer 2: ```suggestion
It is recommended that triage and remediation be prioritized for the top projects listed in this table, especially if priority filters are enabled.
```

Author: not sure about the "be" form of the verb, but yes to correcting my preposition.
